packages/dataviews/src/types.ts

```diff
@@ -103,17 +97,34 @@ export interface Field< Item extends AnyItem > {
 	 * Filter config for the field.
 	 */
 	filterBy?: FilterByConfig | undefined;
-}
-
-export type NormalizedField< Item extends AnyItem > = Field< Item > &
-	Required< Pick< Field< Item >, 'header' | 'getValue' | 'render' > >;
+} & ( Item extends ItemRecord
+	? {
+			/**
+			 * Callback used to retrieve the value of the field from the item.
+			 * Defaults to `item[ field.id ]`.
+			 */
+			getValue?: ( args: { item: Item } ) => any;
+	  }
+	: {
+			/**
+			 * Callback used to retrieve the value of the field from the item.
+			 * Defaults to `item[ field.id ]`.
```

Comment:
Is the description of the default correct here? It looks like the default is `item => item.id`. Reading this documentation, it looks like it's configurable by a field option?

https://github.com/WordPress/gutenberg/blob/89ee50d5953afc5e9651e7aca2e5afa92915326b/packages/dataviews/src/dataviews.tsx#L49

Reply:
Oh, never mind. I see I was confusing the id with the value 🤦

Reply:
Yes, there is `item.id` and also `field.id` 🙂 I wonder if it would be possible to typecheck that the `Item` really has the field read by the `Field` type. Something like:
```js
type Field<Item extends ItemWithField<FieldName>, FieldName> = {
  id: FieldName;
  getValue: ( item: Item ) => unknown;
}
```
which would fail if the `Item` and `FieldName` types don't match.